cpp/dolfin/CMakeLists.txt

```diff
@@ -86,8 +86,15 @@ endif()
 # UFC
 target_include_directories(dolfin SYSTEM PUBLIC ${UFC_INCLUDE_DIRS})
 
+# Currently only used for Eigen
+set(DOLFIN_ALIGNMENT_BYTES "32" CACHE STRING "\
+Minimum alignment in bytes used for numeric arrays. \
```

Comment:
Min of max? Eigen name has `MAX` in it.

Reply:
Yes, this is weird. Eigen docs say "EIGEN_MAX_ALIGN_BYTES - Must be a power of two, or 0. Defines an upper bound on the memory boundary in bytes on which dynamically and statically allocated data may be aligned by Eigen."
Why would max make sense in this context? As stated in my latest comment below, Eigen uses an alignment in bytes which is **at least** this big. If `-march` requires a stricter (i.e. larger) alignment, this is used instead. Therefore I wrote "minimum" in the explaining comment.

Reply:
I found the naming `EIGEN_MAX_ALIGN_BYTES` confusing too when I read the Eigen documentation but I thought I was being stupid :octocat:

Reply:
It may make sense from their internal use:
```
#if EIGEN_IDEAL_MAX_ALIGN_BYTES > EIGEN_MAX_ALIGN_BYTES
    #define EIGEN_DEFAULT_ALIGN_BYTES EIGEN_IDEAL_MAX_ALIGN_BYTES
#else
    #define EIGEN_DEFAULT_ALIGN_BYTES EIGEN_MAX_ALIGN_BYTES
#endif
```
So `EIGEN_MAX_ALIGN_BYTES` can be interpreted something like `EIGEN_MAX_USER_REQUESTED_ALIGN_BYTES` but still... it's weird...

Reply:
@w1th0utnam3 Could you put some text to this effect in the comment? Otherwise we'll all forget this and be rehashing it again in 12 months.

Reply:
@garth-wells Yes, that's reasonable. Regarding the comment: unfortunately, the CMake console gui uses only a single line to display the comment. The comment for the explanation will obviously be longer than most terminal widths. Do you think this is ok?
![image](https://user-images.githubusercontent.com/12967904/44088030-ff2d8476-9fc1-11e8-9af3-9f7b713e366e.png)
Full comment:
```
set(DOLFIN_EIGEN_MAX_ALIGN_BYTES "32" CACHE STRING "\
Sets memory alignment of Eigen matrices. Can be used for compatibility with \
user code that is compiled with AVX2 or AVX-512 vector instructions. \
Note that if an architecture flag (e.g. \"-march=skylake-avx512\") is \
set for DOLFIN, Eigen will use the appropriate ideal alignment instead if \
it is stricter. Otherwise, the value of this variable will be used by Eigen \
for the alignment of all arrays/matrices.\
")
```
At least it's documented if you look into CMakeLists.txt or the cache...

Reply:
CMake help string looks good, but I had in mind a developer 'code' comment on max vs min so in 12 months when we all forget we can be reminded of your reasoning.